### ISSUE TITLE
Document the effect of `mutual` to the order of type checking

### ISSUE THREAD

**Issue by JLimperg** [agda/agda#3926]
The following code is rejected because `tt` is not in scope in the type of `prf`:

```agda
mutual
  data Unit : Set where
    tt : Unit

  P : Unit → Set
  P x = Unit

  prf : P tt
  prf = tt
```

Without the `mutual` or with an implicit mutual block, everything is fine. Tested with Agda 2.6.0.1 and HEAD.

**Comment by jespercockx** [comment]
Maybe something to do with @andreasabel 's recent refactoring of the implementation of mutual blocks?

**Comment by andreasabel** [comment]
Why is this a bug?  `mutual` instructs Agda to first check the signatures, then the implementations of the declarations in the mutual block.  Then, of course, `tt` is not in scope when checking the signature of `prf`.

**Comment by andreasabel** [comment]
This has been Agda's behavior at least since Agda 2.4.2.
```
$ agda-2.4.2.4 Issue3926.agda 
Checking Issue3926 (/Users/abel/play/agda/bugs/Issue3926.agda).
/Users/abel/play/agda/bugs/Issue3926.agda:8,11-13
Not in scope:
  tt at /Users/abel/play/agda/bugs/Issue3926.agda:8,11-13
when scope checking tt
```
I am closing this issue, feel free to reopen should there be need for further discussion or maybe improvement of the documentation.

**Comment by JLimperg** [comment]
Ah, so this is expected behaviour. I was confused for two reasons:

- I thought that if a block of definitions typechecks without `mutual`, then it should also typecheck with `mutual`.
- I thought that if a block typechecks as an implicit mutual block, then it should also typecheck as an explicit mutual block.

More practically, the docs for `mutual` would benefit from adding your explanation that all signatures are checked before all bodies, including those of data types.

**Comment by andreasabel** [comment]
TODO: improve documentation according to @JLimperg 's suggestions.

**Comment by UlfNorell** [comment]
Fixed by #5180
